src/components/SignUpProfile/SignUpProfile.jsx

```diff
@@ -0,0 +1,82 @@
+import * as React from "react"
+import { useState } from "react"
+import { Link, useNavigate } from "react-router-dom"
+import { Button, Card, Alert } from "react-bootstrap" 
+import { doc, updateDoc } from "firebase/firestore"
+import { useAuth } from "../../contexts/AuthContext"
+import { db } from "../../firebase"
+import "./SignUpProfile.css"
+
+export default function SignUpProfile(props) {
+    // get user data from the database
+    const { currentUser } = useAuth()
+    const navigate = useNavigate()
+    const [loading, setLoading] = useState(false)
+    const diets = ["Gluten Free", "Ketogenic", "Vegetarian", "Vegan", "Pescetarian", "Paleo"]
+    const [dietChecked, setDietChecked] = useState([false, false, false, false, false, false])
+    const allergies = ["Diary", "Peanut", "Soy", "Egg", "Shellfish", "Tree Nut", "Gluten"]
```

Comment:
As these lists grow, you may want to consider more scalable UI components for this job.

Reply:
Maybe something like the "Public Methods" component here: https://ericgio.github.io/react-bootstrap-typeahead/#public-methods